### ISSUE TITLE
Unable to create custom HeadersList

### ISSUE THREAD

**Issue by adrianjagielak** [b5i/YouTubeKit#2]
The README may be outdated. The following code was directly copied from the README and it doesn't compile:

<img width="1074" alt="Zrzut ekranu 2023-09-5 o 20 17 57" src="https://github.com/b5i/YouTubeKit/assets/21083675/453224b4-e6d3-4cc9-ad42-d673d22706d6">

Code:
```swift
let myCustomHeadersFunction: () -> HeadersList = {
    HeadersList(
        url: URL(string: "https://www.myrequesturl.com")!,
        method: .POST,
        headers: [
            .init(name: "Accept", content: "*/*"),
            .init(name: "Accept-Encoding", content: "gzip, deflate, br"),
            .init(name: "Accept-Language", content: "\(YTM.selectedLocale);q=0.9"),
        ],
        addQueryAfterParts: [
            .init(index: 0, encode: true)
        ],
        httpBody: [
            "my query is: ",
            " and it is really cool!"
        ]
    )
}
```

Errors:
```
Extra arguments at positions #1, #2, #3, #4, #5 in call
Missing argument for parameter 'from' in call
Cannot infer contextual base in reference to member 'POST'
'Any' cannot be constructed because it has no accessible initializers
'Any' cannot be constructed because it has no accessible initializers
'Any' cannot be constructed because it has no accessible initializers
'Any' cannot be constructed because it has no accessible initializers
```

**Comment by b5i** [comment]
Hello, thanks for pointing out this issue. It seems that all the structs of the package need their default initialiser explicitly declared to be used outside the package itself. I wasn't facing this issue before because the import of YouTubeKit was made with the `@testable` modifier like this `@testable import YouTubeKit`. I'm going to declare all those initialisers as soon as possible. Thanks again

**Comment by b5i** [comment]
Ok it should be fixed with the last commit, `fac7c95`. Tell me if you face any more issue 🚀

**Comment by adrianjagielak** [comment]
Thank you for the quick fix! I checked and it seems to be working fine now.
